addons/mrp/models/mrp_production.py

```diff
@@ -1289,7 +1289,15 @@ def _link_workorders_and_moves(self):
                 previous_workorder = workorder
                 last_workorder_per_bom[workorder.operation_id.bom_id] = workorder
         for move in (self.move_raw_ids | self.move_finished_ids):
-            if move.operation_id:
+            if move.operation_id and not move.operation_id.active:
+                bom_line = move.bom_line_id
+                raise ValidationError(_(
+                        'The operation %(operation_name)s is archived and it is still used '
+                        'as a consumed operation in the component of %(active_bom_line)s.',
+                        operation_name=bom_line.operation_id.name,
+                        active_bom_line=bom_line.product_id.name,
+                    ))
+            elif move.operation_id.active:
```

Comment:
```suggestion
            elif move.operation_id:
```

Reply:
@yelizariev Changes have been applied!

Thank You!